src/itensor.jl

```diff
@@ -1140,15 +1140,16 @@ A[i => 1, i' => 2] # 2.0, same as: A[i' => 2, i => 1]
 @propagate_inbounds (getindex(T::ITensor, ivs::Vararg{Any,N})::Any) where {N} =
   _getindex(tensor(T), ivs...)
 
+## Allowing one to get the first ITensor element if its an order 0 tensor or an order 1 tensor with a dimension of 1. Also convert GPU back to CPU
 @propagate_inbounds function getindex(T::ITensor)::Any
-  if order(T) != 0
-    throw(
+  if order(T) != 0 && dim(T) != 1
+    ITensors.throw(
       DimensionMismatch(
         "In scalar(T) or T[], ITensor T is not a scalar (it has indices $(inds(T)))."
       ),
     )
   end
-  return tensor(T)[]
+  return NDTensors.cpu(tensor(T))[]
```

Comment:
Can we fix this in NDTensors directly and dispatch `getindex(T::Tensor)` on `leaf_parenttype`, and transfer to CPU if it is `CuArray`? Or just call `cpu` in `getindex(T::Tensor)` in `NDTensors`?